server/src/main/java/org/elasticsearch/cluster/desirednodes/DesiredNodesSettingsValidator.java

```diff
@@ -23,19 +23,26 @@
 public class DesiredNodesSettingsValidator {
     private record DesiredNodeValidationError(int position, @Nullable String externalId, RuntimeException exception) {}
 
+    @Deprecated(forRemoval = true)
+    private static boolean isCurrentIgnoringQualifiers(String expected, String actual) {
+        // The expected version string may contain -SNAPSHOT; the actual may not
+        // If the actual contains it, the expected must match entirely; this means actual must be a prefix for expected
+        return Strings.isNullOrEmpty(actual) == false && expected.startsWith(actual);
```

Comment:
Could use `hasText` here so we don't have the negation

Reply:
TIL, thanks